server/pairing/payload_mounter.go

```diff
@@ -0,0 +1,339 @@
+		payloadLoader:            apl,
+	}, nil
+}
+
+// Mount loads and prepares the payload to be stored in the AccountPayloadLoader's state ready for later access
+func (apm *AccountPayloadMounter) Mount() error {
+	l := apm.logger.Named("Mount()")
+	l.Debug("fired")
+
+	err := apm.payloadLoader.LoadFromSource()
+	if err != nil {
+		return err
+	}
+	l.Debug("after LoadFromSource")
+
+	pb, err := apm.accountPayloadMarshaller.MarshalProtobuf()
+	if err != nil {
+		return err
+	}
+	l.Debug(
+		"after MarshalProtobuf",
+		zap.Any("accountPayloadMarshaller.accountPayloadMarshaller.keys", apm.accountPayloadMarshaller.keys),
+		zap.Any("accountPayloadMarshaller.accountPayloadMarshaller.multiaccount", apm.accountPayloadMarshaller.multiaccount),
+		zap.String("accountPayloadMarshaller.accountPayloadMarshaller.password", apm.accountPayloadMarshaller.password),
+		zap.Binary("pb", pb),
+	)
+
+	return apm.encryptor.encrypt(pb)
+}
+
+func (apm *AccountPayloadMounter) ToSend() []byte {
+	return apm.encryptor.getEncrypted()
+}
+
+func (apm *AccountPayloadMounter) LockPayload() {
+	apm.encryptor.lockPayload()
+}
+
+// AccountPayloadLoader is responsible for loading, parsing and validating AccountPayload data
+type AccountPayloadLoader struct {
+	*AccountPayload
+
+	multiaccountsDB *multiaccounts.Database
+	keystorePath    string
+	keyUID          string
+}
+
+func NewAccountPayloadLoader(p *AccountPayload, config *SenderConfig) (*AccountPayloadLoader, error) {
+	ppr := &AccountPayloadLoader{
+		AccountPayload: p,
+	}
+
+	if config == nil {
+		return ppr, nil
+	}
+
+	ppr.multiaccountsDB = config.DB
+	ppr.keyUID = config.KeyUID
+	ppr.password = config.Password
+	ppr.keystorePath = config.KeystorePath
+	return ppr, nil
+}
+
+func (apl *AccountPayloadLoader) LoadFromSource() error {
+	err := apl.loadKeys(apl.keystorePath)
+	if err != nil {
+		return err
+	}
+
+	err = validateKeys(apl.keys, apl.password)
+	if err != nil {
+		return err
+	}
+
+	apl.multiaccount, err = apl.multiaccountsDB.GetAccount(apl.keyUID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (apl *AccountPayloadLoader) loadKeys(keyStorePath string) error {
+	apl.keys = make(map[string][]byte)
+
+	fileWalker := func(path string, fileInfo os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if fileInfo.IsDir() || filepath.Dir(path) != keyStorePath {
+			return nil
+		}
+
+		rawKeyFile, err := ioutil.ReadFile(path)
+		if err != nil {
+			return fmt.Errorf("invalid account key file: %v", err)
+		}
+
+		accountKey := new(keystore.EncryptedKeyJSONV3)
+		if err := json.Unmarshal(rawKeyFile, &accountKey); err != nil {
+			return fmt.Errorf("failed to read key file: %s", err)
+		}
+
+		if len(accountKey.Address) != 40 {
+			return fmt.Errorf("account key address has invalid length '%s'", accountKey.Address)
+		}
+
+		apl.keys[fileInfo.Name()] = rawKeyFile
+
+		return nil
+	}
+
+	err := filepath.Walk(keyStorePath, fileWalker)
```

Comment:
Most definitely not a problem in this context, but I felt like sharing, from the Go doc says:

> Walk is less efficient than WalkDir, introduced in Go 1.16, which avoids calling os.Lstat on every visited file or directory. 
>
> -- https://cs.opensource.google/go/go/+/go1.20.2:src/path/filepath/path.go;l=566

Should we just use `WalkDir` in general? The doc only says the upside, not any downside, so I'm left wondering if it's always better. Reference for the the other `WalkDirFunc` https://cs.opensource.google/go/go/+/go1.20.2:src/io/fs/walk.go;l=69

Reply:
Hmm, this is a good point. `WalkDir` was introduced recently I've not even considered its usage. Oh I just did a Google search and it was released 2 years ago 😆. `WalkDir` seems to be superior. I'll update it.

Buuuut, now I have a question! This comment you've made is an unusually insightful suggestion and question and requires a large degree of context, how were you able to give such feedback?

Reply:
> how were you able to give such feedback?

Oh that's a great question. I'm trying to learn Go for more than just closing issues, I'm assuming I know nothing, even though we all know programming languages have a lot of overlap. I really like that many Go libraries are reasonably well documented, they point out interesting points, not just *what* functions do. But in the end, curiosity? I find it incredibly satisfying to read API docs and jumping to source every now and then. Ah the beauty of open-source!